### PR TITLE
SameSite Config Option for Cookie

### PR DESCRIPTION
**Pull request by JVickery-TBS**
Adds `ckanext.csrf_filter.same_site` config option which defaults to `'None'` and uses it in setting the token cookie. Asserts the allowed values of 'Strict', 'Lax', and 'None'

This should solve the issue: https://github.com/qld-gov-au/ckanext-csrf-filter/issues/28

### CODE REVIEW

**Review by ThrawnCA — APPROVED** [review]
Looks great, thanks for the contribution!

Could I get you to target the 'develop' branch instead of 'main'? That's used in our development environments to try features out.

File: ckanext/csrf_filter/anti_csrf.py [hunk 1]
```diff
@@ -75,6 +77,10 @@ def configure(config):
         LOG.warning("Site %s is not secure! CSRF tokens may be exposed!", site_url)
         secure_cookies = False
 
+    same_site = config.get('ckanext.csrf_filter.same_site', 'None')
```
Comment (ThrawnCA) [inline ckanext/csrf_filter/anti_csrf.py:80]: This could subsume the existing exclusion of the API.

**Comment by duttonw** [comment]
Hi @JVickery-TBS, Due to github cost cutting, they disable github actions on forked repo's. Can you enable github actions on your fork and retrigger a build so cicd pipeline can run letting us know if tests pass, etc. https://github.com/open-data/ckanext-csrf-filter/actions

**Comment by JVickery-TBS** [comment]
@duttonw Workflows enabled on our fork now. Did you want me to add on dispatch and on PR to the test yaml?

**Comment by JVickery-TBS** [comment]
@duttonw @ThrawnCA oops sorry I have more code that made it into this PR. Regarding the option for custom regex rules to exempt from token checks.

I will clean this PR up and make a new one for the custom regex rules stuff. And then we will get the tests running on those.

File: ckanext/csrf_filter/anti_csrf.py [hunk 2]
```diff
@@ -90,6 +96,13 @@ def configure(config):
     token_expiry_age = 60 * config.get('ckanext.csrf_filter.token_expiry_minutes', 30)
     token_renewal_age = 60 * config.get('ckanext.csrf_filter.token_renewal_minutes', 10)
 
+    exempt_rules = config.get('ckanext.csrf_filter.exempt_rules', None)
+    if exempt_rules is not None:
```
Comment (ThrawnCA) [inline ckanext/csrf_filter/anti_csrf.py:100]: This could be shortened to something like:

    if exempt_rules:
        exempt_rules = [re.compile(rule) for rule in exempt_rules.split()]

And it should probably be in a separate pull request.

File: ckanext/csrf_filter/plugin.py [hunk 3]
```diff
@@ -95,6 +96,10 @@ def check_csrf():
         def set_csrf_token(response):
             """ Apply a CSRF token to all response bodies.
             """
+            if hasattr(response, 'response') \
```
Comment (ThrawnCA) [inline ckanext/csrf_filter/plugin.py:99]: Couldn't this be done in one step using the default value of `getattr`?

**Comment by JVickery-TBS** [comment]
Just splitting this up into multiple PRs now.

https://github.com/qld-gov-au/ckanext-csrf-filter/pull/36

**Comment by JVickery-TBS** [comment]
And this one: https://github.com/qld-gov-au/ckanext-csrf-filter/pull/37 (includes the `getattr` feedback from this conversation)

**Comment by JVickery-TBS** [comment]
And finally this one: https://github.com/qld-gov-au/ckanext-csrf-filter/pull/38
